util/fibers/uring_socket.cc

```diff
@@ -47,17 +50,52 @@ UringSocket::~UringSocket() {
   LOG_IF(WARNING, ec) << "Error closing socket " << ec << "/" << ec.message();
 }
 
+error_code UringSocket::Create(unsigned short protocol_family) {
+  error_code ec = LinuxSocketBase::Create(protocol_family);
+  if (ec) {
+    return ec;
+  }
+
+  UringProactor* proactor = GetProactor();
+  CHECK(proactor && is_direct_fd_ == 0);
+  DCHECK(proactor->InMyThread());
+
+  if (proactor->HasDirectFD()) {
```

Comment:
currently this is the only function that actually uses directfd just because accept_server calls it to create a listening socket. This is how I limit its impact on the rest of the code.